docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+If the hard fork combinator were to always agree with the ledger---thereby avoiding the awkward surprise arising possible after a Chain Growth violation---then it would have to count only slots as the ledger does and not blocks.
+As a result, `GetEraStart` and `GetInterpreter` would no longer exhibit weak monotonicity.
+Specifically, a particular Chain Growth violation could cause `GetEraStart` to yield `Just x` in one moment and then `Nothing` or `Just y` where `x /= y` at a later moment, despite the node never rolling back more than k blocks.
+(`GetInterpreter` would be more intricate to describe, but the nub is exactly the same.)
+
+It is remarkable that every query without a marker is not weakly monotonic, and no one expects them to be!
+For example, there's no surprise that the UTxO changes when the node switches chains.
+However, it was decided that `GetEraStart` and `GetInterpreter` needed to exhibit this monotonicity because their replies include information about _time_.
+For example, if a user needed to know at what UTC time a specific slot would begin, the Cardano CLI would determine that using the `GetInterpreter` query.
+And because `GetInterpreter` is weakly monotonic, the Cardano CLI will reply with "unable to translate that slot to a UTC time" until the UTC time is both predictable from the ledger state answering the query and moreover guaranteed to be the same for every ledger state the node could subsequently select unless it somehow rolls back more than k blocks.
+This decision may seem even more reasonable when you consider that the opposite question is handled the same exact way: the node will not let the user translate a UTC time to a slot until that answer is also k-rollback-monotonic.
+(Frisby's inference is that UTC time seems so fundamental that it's easy to wrongly assume that the translation back and forth between slots and UTC times does not depend on the evolution of the chain.)
+Requiring this monotonicity for `GetEraStart` and `GetInterpreter` avoids that confusion from manifesting.
+
+## Proposal: Best of Both Worlds
+
+Consider the following alteration of the hard fork combinator.
+
+- Remove the hard fork combinator's second trigger constraint, such that it transitions to the next era exactly when the ledger governance updates the major version of the protocol version protocol parameter (mutatis mutandi as of Conway).
+- Only answer `GetEraStart` and `GetInterpreter` queries using an immutable ledger state.
+- (The ledger rules could then return the voting deadline to 7k/f; one stability window before the epoch transition would be sufficient again.)
+
+This would have the exact same answers to Question 1 and Question 2 as does the simplified node, and the Note-To-Client time translation queries would still exhibit k-rollback-monotonicity.
+
+There are three main benefits for the Consensus Team.
+
+- The hard fork combinator would no longer need the block-counting logic.
```

Comment:
Does this mean that, when checking era transitions we'd be counting slots instead?